pkg/services/object/get/assemble.go

```diff
@@ -163,12 +163,15 @@ func (exec *execCtx) overtakePayloadDirectly(children []oid.ID, rngs []objectSDK
 			r = &rngs[i]
 		}
 
-		child, ok := exec.getChild(children[i], r, !withRng && checkRight)
+		child, reader, ok := exec.getChildStream(children[i], r, !withRng && checkRight)
 		if !ok {
+			if reader != nil {
```

Comment:
i would not expect this possible. Better to have `reader` and `ok` mutually exclusive. Otherwise all callers will have to do this. Method can close opened reader on failure itself